pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -1138,15 +1144,13 @@ func (m *kubeGenericRuntimeManager) computeInitContainerActions(pod *v1.Pod, pod
 					if liveness == proberesults.Failure {
 						// If the restartable init container failed the liveness probe,
 						// restart it.
-						changes.ContainersToKill[status.ID] = containerToKillInfo{
-							name:      container.Name,
-							container: container,
-							message:   fmt.Sprintf("Init container %s failed liveness probe", container.Name),
-							reason:    reasonLivenessProbe,
-						}
-						changes.InitContainersToStart = append(changes.InitContainersToStart, i)
+						message = fmt.Sprintf("%s failed liveness probe, will be restarted", message)
+						reason = reasonLivenessProbe
+						restartContainer = true
+						break
```

Comment:
this is a new break statement. Is it intentionally placed here? It looks like it wouldn't change the flow. But it's not easy to tell